hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncConnectionImpl.java

```diff
@@ -394,4 +394,7 @@ public Hbck getHbck(ServerName masterServer) throws IOException {
   Optional<MetricsConnection> getConnectionMetrics() {
     return metrics;
   }
+
+  @VisibleForTesting
+  AsyncRegistry getRegistry() { return registry; }
```

Comment:
Hmm. Fine for master and branch-2, but note a headache here for branch-1 (if a backport is desired). The precursor implementation in branch-1 is ClusterRegistry. AsyncRegistry was a big refactor via HBASE-16835. Not suggesting this needs be different, and some simple substitutions may get you most of the way there, but you may want to look at this.